### PR TITLE
Add NUCLEO_F429ZI + NUCLEO_L476RG support

### PR DESCRIPTION
**Pull request by adustm**
Hello,
I've been able to use this test with NUCLEO_F429ZI and NUCLEO_L476RG, together with an ESP8266-01 wifi module
```
Connecting...
Success

MAC: 18:fe:34:13:c1:9e
IP: 192.168.1.100
Netmask: (null)
Gateway: (null)
RSSI: 0

Sending HTTP request to www.arm.com...
sent 38 [GET / HTTP/1.1]
recv 64 [HTTP/1.1 200 OK]

Done
```

Is it possible to add this modification to main.cpp + Readme file, please ?
@bcostm

### CODE REVIEW

**Comment by adustm** [comment]
Hello @adbridge, @0xc0170 
Do you know who would be able to add this PR to this repo ?

Kind regards

**Comment by adbridge** [comment]
@bulislaw please review/merge

File: main.cpp [hunk 1]
```diff
@@ -20,12 +20,24 @@
...
 #include "OdinWiFiInterface.h"
 OdinWiFiInterface wifi;
+
+#elif TARGET_NUCLEO_L476RG
+#include "ESP8266Interface.h"
+// ESP8266_WIFI TX to be connected to D2 (Serial1_Rx)
+// ESP8266_WIFI_RX to be connected to D8 (Serial1_Tx)
+ESP8266Interface wifi(D8, D2);
+
+#elif  TARGET_NUCLEO_F429ZI
```
Comment (bulislaw) [inline main.cpp]: Could we not have special case for this board if the used pins are the same as for 'default' esp, it should work OOB if you use D1/D0.

File: README.md [hunk 2]
```diff
@@ -11,6 +11,8 @@ The program brings up the WiFi and the underlying network interface, and uses it
 ### Supported hardware ###
 
 * [UBLOX Odin board](https://developer.mbed.org/platforms/ublox-EVK-ODIN-W2/) (`UBLOX_EVK_ODIN_W2` target when using mbed CLI)
+* [NUCLEO_F429ZI](https://developer.mbed.org/platforms/ST-Nucleo-F429ZI/) with ESP8266-01 module
```
Comment (bulislaw) [inline README.md]: Could you also add a note about used pins here please.

File: main.cpp (hunk 1 above)
Comment (MarceloSalazar) [inline main.cpp]: @bulislaw can we add the pin config to the mbed_app.json, as it's done for the Client example?

Comment (bulislaw) [inline main.cpp]: We can, I'll be happy to accept PR :)

Comment (adustm) [inline main.cpp]: D0/D1 can work for NUCLEO_F429ZI. 
Unfortunately D0 / D1 are connected to the USB-debug-serial on NUCLEO_L476RG . 
We have to use some other pins.

@MarceloSalazar do you know how to do mbed_apps.json modification ? I did not find it.

Comment (adustm) [inline main.cpp]: I must add that the !TARGET_FF_ARDUINO did not work for NUCLEO platforms. Do you know why ?

File: README.md (hunk 2 above)
Comment (adustm) [inline README.md]: ok

File: main.cpp (hunk 1 above)
Comment (MarceloSalazar) [inline main.cpp]: > @MarceloSalazar do you know how to do mbed_apps.json modification ? I did not find it.

You can see the mbed_app.json from the Client example here:
https://github.com/ARMmbed/mbed-os-example-client/blob/master/mbed_app.json#L21

**Comment by adustm** [comment]
Hi @bulislaw 
I think it fits your need this time.
Let me know

File: main.cpp (hunk 1 above)
Comment (adustm) [inline main.cpp]: Thanks @MarceloSalazar

**Comment by 0xc0170** [comment]
Thanks @adustm , LGTM

**Comment by bulislaw** [comment]
Thanks @adustm LGTM

**Pull request merged**